### ISSUE TITLE
GREATS method with gradient accumulation

### ISSUE THREAD

**Issue by millix19** [Jiachen-T-Wang/GREATS#6]
Hi Jiachen, i noticed during GREATS selection in `compute_GradProd_GC_per_iter` function there is call to `optimizer.zero_grad()`. Will this not disrupt the gradient accumulation process? would be great if you could help explain thanks!
